### ISSUE TITLE
Problem stopping the charge session during Cable Check

### ISSUE THREAD

**Issue by IonelFratila** [Sevenstax/FreeV2G#20]
Hello @IonelFratila ,

in the charge loop you can set the response code to either "3: Stop charging" or "4: Renegotiate", see chapter 17.10.17 Set Charge Loop Parameters.

In Pre Charge and Cable Check you can stop by using the stop command 0x43. This forces a TCP disconnect. Do you have any issues with that in the pre charge and charge loop phase? Does the EV needs the EVSE_Shutdown status code?

BR Jan

_Originally posted by @jpo-stx in https://github.com/Sevenstax/FreeV2G/issues/14#issuecomment-978069904_

**Comment by IonelFratila** [comment]
Hi @jpo-stx,
Because my last message was lost from the forum, I will resend this message:

Regarding your answer about the possibility to send the stop request to the EV, in case the user ask for normal stop of the charging session:
1.In the charge loop I tried the response code "3.Stop charging" in the Set Charge Loop Parameters command 0x70 and everything seems to be OK, I can see the normal message sequence between EVSE and EV:
-Current Demand Request (last message send by the EV)
-Current Demand Response OK (with EVSE_StatusCode=EVSE_ShutDown)
-Power Delivery Request (Power Off)
-Power Delivery Response OK
-Welding Detection Request
-Welding Detection Response OK
-Session Stop Request
-Session Stop Response OK
2.In the Cable Check phase, I tried to send the Stop V2G Service 0x43 command, as you suggested, but the result is that after the last Cable Check  Request from EV no other message is exchanged between WB and EV. The same situation for PreCharge phase.
I also tried to send the result code "2: Unable to deliver EVSE parameters" in the Set Cable Check Parameters 0x6D to WB or to send result code "1: Cable check failed" in the Set Cable Check Status command, but the result is that the message sequence between WB and EV ends as follow:
-Cable Check Request
-Cable Check Respose (Failed) (with EVSE_StatusCode=EVSE_NotReady)
-Session Stop Request
after that EV signals an error and the communication is stopped.

Please tell me if you have other ideas about how to stop the charging session normally (without an error from EV) after Session Start but before entering Charge Loop.

Maybe the same result code "3. Stop charging" and the same behavior of WB as in Set Charge Loop Parameters command can be added also in Set Cable Check Parameters or Set PreCharge Parameters commands (I think this is actually needed for all messages sent from host controller to WB after receiving Session Started message 0x80).

Thank you,
Best Regards,
Ionel Fratila

**Comment by jpo-stx** [comment]
Hi @IonelFratila 

> Please tell me if you have other ideas about how to stop the charging session normally (without an error from EV) after Session Start but before entering Charge Loop.

I do not think that there is another possibility besides the stop command and the response code 2 in the cable check message.

> Maybe the same result code "3. Stop charging" and the same behavior of WB as in Set Charge Loop Parameters command can be added also in Set Cable Check Parameters or Set PreCharge Parameters commands (I think this is actually needed for all messages sent from host controller to WB after receiving Session Started message 0x80).
> 
> Thank you, Best Regards, Ionel Fratila

I think this would be the best solution. We'll add that in a future version!

BR
Jan

**Comment by IonelFratila** [comment]
Hi @jpo-stx ,
Thank You for your prompt reply. I will wait for this new version.
Best Regards,
Ionel Fratila

**Comment by zacharyalutz** [comment]
Greetings @IonelFratila, @jpo-stx:

The response codes "3: Stop charging" or "4: Renegotiate" for Set Charge Loop Parameters are fairly new. Before they existed, the way I have been indicating an EVSE-initiated shutdown has been to assert Control Pilot transition **x2-->x1** by changing the  duty cycle **5%**-->**100%**. 

In the event of an EVSE-initiated *emergency* shutdown, i have been asserting **state F** on the control pilot by changing the duty cycle to **0%**.

**Comment by IonelFratila** [comment]
Hi @zacharyalutz, @jpo-stx,
Of course PWM 5% duty cycle must be turned OFF either to 100% or 0% (CP state X1 or F), depending of the stop charge condition, according to the standards (DIN 70121 and IEC15118).
The real problem is not how to stop the charging process (which can be done in several ways: turning PWM 5% OFF, turning the power rectifiers OFF, turning V2G communication OFF), but how to stop the V2G charging session through HLC communication with EV in the normal way (complying with the requirements of the standards). This means sending to the EV the stop charging notification (and EVSE_StatusCode accordingly) and answering to all V2G messages coming from the EV, until the V2G session is gracefully closed. Otherwise, EV will signal an error by turning on the red light near the charging connector and the charging connector will be released only after up to 30 seconds (depending on the type of EV).

There are some situations when the normal stop message sequence cannot be accomplished, but this is not the case when the session is terminated due to the user's request (anytime after the session starts).

Currently, the WB module provides normal shutdown only during Charge Loop, by using the Result code "3.Stop charging" when the user requests charging shutdown, or by reaching the ChargeComplete condition transmitted by EV. 
This is the reason why I think this result code (and associated behavior in communication with EV) should be implemented for all messages sent from the host controller to WB after Session Started message. So I will wait for the next SW update regarding this issue.

Best Regards,
Ionel Fratila

**Comment by zacharyalutz** [comment]
@IonelFratila: Aha, I understand your concern now. You are *not* concerned with simply stopping the session per se, but rather concerned with conformance to the standard with respect to EVSENotification. 

@jpo-stx @ttu-stx @bbr-stx: I understand that you intend to add the ability for EVSE to initiate a normal shutdown in the next firmware update as you mentioned above, but I am interested in how to manage user-initiated shutdown (which is @IonelFratila's concern) as well as how to manage renegotiation from EVSE-side. To that end, can you please explain the how the host platform can do the following:
  1. Set the values of **EVSENotification** i.e. *StopCharging*, *Renegotiation*, *None*?
  2. Get the values of **ChargeProgress** from EV, i.e. *Start*, *Stop*, *Renegotiate*?    
  3. Get the values of **ResponseCode** for V2G application layer messages e.g *FAILED_ContactorError*, *OK_OldSessionJoined*, etc.

Thanks again,
Zach

**Comment by IonelFratila** [comment]
Hi @jpo-stx ,
I have a new issue with my Skoda CitiGo which I'm using for testing the host controller SW.

During charging loop, EV send Current Demand Request messages and WB replies with Current Demand Response messages and everything seems to be OK. At random moments during charging loop, for no particular reason, EV switch the CP state from C2 to B2 and send a Session Stop Request message. The WB module replies to this message with Session Stop Response (Failed) having ResponseCode=FAILED_SequenceError. To the host controller, WB send only a Session stopped message (ModID=0x27,SubID=0x81), so nothing can be done from the host controller side.
Because this accidental stop of the charging process is very annoying, I need to find a way to resume the charge process without user intervention (without removing and reinserting the connector into the EV). I have noticed that if after stopping PWM (switch from 5% to 100%) I restart the entire process by sending again PWM=5% and restarting the SLAC Service and the V2G Service and then sending Start SLAC Matching Process (0x44), EV send again SLAC_Param.Req, WB answer to this message and the entire SLAC Matching is performed OK (terminated with SLAC_Match.Req from EV and SLAC_Match.Confirm from EVSE, containing NID and NMK parameters). After that, nothing happens until the Session Start timeout (from the standard) expires.
The behavior is the same even if I tried to stop  V2G Service before restarting the SLAC Service. I also noticed that with each new OK run of the SLAC matching process, the confirmation message SLAC_Match.Confirmation sent by WB contains other values ​​for the NID and NMK parameters. According to the standard, the NID and NMK parameters must be initialized before plug-in the EV and should remain unchanged until the EV is plugged out (CP State =A).
I tried also not to restart SLAC Service, but in this case EV is not trying to resume the SLAC Matching process.

Questions:
1.What commands should I send to WB in order to confirm the Session Stop Request message from EV (I think it is not normal for WB to send a "Session Stop Response" message with ResponseCode=Failed to EV)?
2.When the NID and NMK parameters used in the SLAC_Match.Confirmation message to EV are (re)initialized? How can I make the WB to use the same values ​​for these parameters as at the beginning of the charging session (first SLAC Matching process after EV plug-in) (this is also a mandatory requirement of the standard, I don't know why the requirement is not met by WB)?
3.Is it possible that after a successfully completed SLAC Matching process EV does not continue with V2G communication because the NID and NMK parameters differ from the previous values (from the initial SLAC Matching)?
4.How and when the command Join HPGP network (ModID=0x2A, SubID=0x40) can be used? It is the only way I can send particular values for NID and NMK to WB and I figured that I can use it for keeping the same values as in the first SLAC Matching process. The problem is that every time when I send this command I get the result code=0x01 (Module Busy?!), whether the SLAC Service is started or not or if the SLAC Matching process is running or not.
5.You have any other ideas I can try in order to resume the charging process, without plugging out the EV?

I'm using a CCS communication analyzer from Vector to monitor the communication between EVSE and EV and also the host controller creates a log with the messages exchanged with the WB module. I have attached the log files from both sources. I hope it will be helpful to be able to analyze the problem in detail. The log files from CCS analyzer are saved in *.csv format.
*  20211214_Skoda CitiGo accidental Stop.txt - log file containing WB messages, starting from the moment of interruption and including 2 attempts to resume the process (SLAC Matching OK but not followed by Session Started)
* 20211214_Skoda CitiGo normal Start.txt - log file containing WB messages for a normal charging session start (from the WB initialization to Cable Check start) - attached for comparison only
* 20211214_Accidental stop charge.csv - log file from Vector analyzer, containing the entire charging process, until accidental stop
* 20211214_SLAC resume but without SessionStart.csv - log file from Vector analyzer, containing the communication between WB and EV after the moment of accidental stop

Thank You very much for help,
Best Regards,
Ionel Fratila

[20211214_Accidental stop charge.csv](https://github.com/Sevenstax/FreeV2G/files/7710994/20211214_Accidental.stop.charge.csv)
[20211214_Skoda CitiGo accidental Stop.txt](https://github.com/Sevenstax/FreeV2G/files/7710995/20211214_Skoda.CitiGo.accidental.Stop.txt)
[20211214_Skoda CitiGo normal Start.txt](https://github.com/Sevenstax/FreeV2G/files/7710996/20211214_Skoda.CitiGo.normal.Start.txt)
[20211214_SLAC resume but without SessionStart.csv](https://github.com/Sevenstax/FreeV2G/files/7710997/20211214_SLAC.resume.but.without.SessionStart.csv)

**Comment by jpo-stx** [comment]
Hi @IonelFratila , please give me some time to go through this.

**Comment by jpo-stx** [comment]
> @IonelFratila: Aha, I understand your concern now. You are _not_ concerned with simply stopping the session per se, but rather concerned with conformance to the standard with respect to EVSENotification.
> 
> @jpo-stx @ttu-stx @bbr-stx: I understand that you intend to add the ability for EVSE to initiate a normal shutdown in the next firmware update as you mentioned above, but I am interested in how to manage user-initiated shutdown (which is @IonelFratila's concern) as well as how to manage renegotiation from EVSE-side. To that end, can you please explain the how the host platform can do the following:
> 
>     1. Set the values of **EVSENotification** i.e. _StopCharging_, _Renegotiation_, _None_?

It will be possible to trigger stop charging and renegotiation in the cable check, pre charge and charge loop phases.

>     2. Get the values of **ChargeProgress** from EV, i.e. _Start_, _Stop_, _Renegotiate_?

You get these indirectly:
Start - Request Start Charging
Stop - Request Stop Charging
Renegotiate - Since this is handled internally, you just need to provide the schedules.

>     3. Get the values of **ResponseCode** for V2G application layer messages e.g _FAILED_ContactorError_, _OK_OldSessionJoined_, etc.

We can add this as an improvement, to provide the response code in the session stopped notification.

> 
> Thanks again, Zach

**Comment by IonelFratila** [comment]
Hi @jpo-stx,
I found on the CODICO website the new version of SW for the WB module, v.01_01_06 and the associated White Beet - User manual revision no.2.10 and I am a little confused about them.
In the ReleaseNotes document regarding v.01_01_06, at Chapter 6 it is said that one of the new features added is related to "Allow stop charging and renegotiation in cable check and precharging phase" which we discussed. But in the User Manual rev.no.2.10 I can found nothing about this subject. In fact, the only changes in this document are related to EV Port mirroring and a note regarding Set Energy transfer Modes command.
1. Can you confirm that the User Manual rev.no.2.10 is referring to WB V.01_01_06 (nothing is written in the doc file about the WB version it refers to) ? Can you help me to understand how the new feature "stop charging and renegotiation in cable check and precharging phase" can be used?

2. Regarding my previous questions about charge resuming (NID and NMK unchaged until CPState=A), do you have any news? Do you think that it is possible to resume charging this way? Do you have other ideas about not resuming communication with EV after a successful SLAC?

Thank You,
Best Regards,
Ionel Fratila

**Comment by bbr-stx** [comment]
Hi @IonelFratila, I just updated the Whitebeet Manual (2.11) in GitHub. There you will find the changes regarding stopping the charge in the cable check and precharge phase.

**Comment by IonelFratila** [comment]
Hi @jpo-stx , @bbr-stx ,
I tested the new version 01_01_06 of the WB module, but it seems to be a problem with the implementation of the "3. Stop Charging" result code for the commands Set Cable Check Parameters or Set PreCharge Parameters.
When I try to send Set Cable Check Parameters with Stop Charging result code, I get the result 0xFF (Internal Error) and when I send Set PreCharge Parameters with Stop Charge result code, I get the result 0x04 (Malformed data). Both commands are repeated 3 times,with the same result.
Here are the messages exchanged between host controller and WB module:

Check FW version after module update:
ETx: FFFFFFFFFFFF 123456789ABC 6003 00040008 C01041100000 []00C1
ERx: 123456789ABC C49300215136 6003 00040013 C0104110000B [00095630315F30315F3036]C9C1     V.01_01_06

User stop charge during Cable Check:
ERx: 123456789ABC C49300215136 6003 0004000E C02787FF0006 [000006090057]62C1	;Request CableCheck Parameters
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000B C0276D160003 [000000]00C1		;Set CableCheck Parameters (EV parameters accepted)
ERx: 123456789ABC C49300215136 6003 00040009 C0276D160001 [00]D1C1		;Set CableCheck Parameters response (OK)
ERx: 123456789ABC C49300215136 6003 00040009 C02981FF0001 [02]CFC1		;CPState=C
ERx: 333300000002 007DFA03950A 86DD						                        ;Here Manual user Stop is decided
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000A C02944170002 [03E8]00C1				;Set PWM=100%
ERx: 123456789ABC C49300215136 6003 00040009 C02944170001 [00]F7C1
ERx: 123456789ABC C49300215136 6003 0004000E C02787FF0006 [000006090057]62C1	;Req CableCheck Parameters
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000B C0276D180003 [030000]00C1		;Set CableCheck Parameters (Stop charging)
ERx: 123456789ABC C49300215136 6003 00040009 C0276D180001 [FF]CFC1		;Set CableCheck Parameters response (Internal Error)
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000B C0276D190003 [030000]00C1		;repeat Set CableCheck Parameters (Stop charging)
ERx: 123456789ABC C49300215136 6003 00040009 C0276D190001 [FF]CEC1		;Set CableCheck Parameters response (Internal Error)
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000B C0276D1A0003 [030000]00C1		;repeat Set CableCheck Parameters (Stop charging)
ERx: 123456789ABC C49300215136 6003 00040009 C0276D1A0001 [FF]CDC1		;Set CableCheck Parameters response (Internal Error)                         
ERx: 4E10 4000C000 123456789ABC C49300215136 6003 00040009 C02981FF0001 [01]D0C1		;CPState=B (due to PWM change)
ERx: 333300000002 007DFA03950A 86DD
ERx: 123456789ABC C49300215136 6003 00040008 C02781FF0000 []D4C1 			;Session Stopped

User stop charge during PreCharge:
ERx: 123456789ABC C49300215136 6003 00040014 C02788FF000C [00000608000D43FF000AFF57]02C1		;Request PreCharge Parameters                                                        
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000E C0276E320006 [0000021770FD]00C1			;Set PreCharge Parameters
ERx: 123456789ABC C49300215136 6003 00040009 C0276E320001 [00]B4C1		;Set PreCharge Parameters confirmation (OK)
                                                                            			;Here Manual user Stop is decided                                                                          
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000A C02944330002 [03E8]00C1				;Set PWM=100%
ERx: 123456789ABC C49300215136 6003 00040009 C02944330001 [00]DBC1
ERx: 123456789ABC C49300215136 6003 00040014 C02788FF000C [00000608000D43FF000AFF57]02C1		;Request PreCharge Parameters
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000E C0276E340006 [0300021770FD]00C1			;Set PreCharge Parameters (Stop Charging)
ERx: 123456789ABC C49300215136 6003 00040009 C0276E340001 [04]AEC1		;Set PreCharge Parameters response (Malformed data)
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000E C0276E350006 [0300021770FD]00C1			;repeat Set PreCharge Parameters (Stop Charging)
ERx: 123456789ABC C49300215136 6003 00040009 C0276E350001 [04]ADC1		;Set PreCharge Parameters response (Malformed data)
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000E C0276E360006 [0300021770FD]00C1			;repeat Set PreCharge Parameters (Stop Charging)
ERx: 4206 4000C000 123456789ABC C49300215136 6003 00040009 C0276E360001 [04]ACC1		;Set PreCharge Parameters response (Malformed data)
ERx: 123456789ABC C49300215136 6003 00040009 C02981FF0001 [01]D0C1		;CPState=B                                                                     
ERx: CE06 4000C000 123456789ABC C49300215136 6003 00040008 C02781FF0000 []D4C1  		;Session Stopped

Please check the commands when the result code is "3.Stop charging" and see what happens.
Thank You,
Ionel Fratila

**Comment by jpo-stx** [comment]
Hi @IonelFratila , there is the timeout value missing in the message format description. Please have a look at the "Set Charge Loop Parameters" message description and set the timeout parameter as it is described there.

I will update the manual with the correct message format in the next days.

**Comment by IonelFratila** [comment]
Hi @jpo-stx ,
I tested again the implementation of the "3. Stop Charging" result code for the commands Set Cable Check Parameters or Set PreCharge Parameters on version 01_01_06 of the WB module, including also the timeout value. Now the commands are accepted, but the problem regarding normal stop of the charge session still persists between WB and EV. When the "Stop Charging" result code is sent, the following messages are exchanged between WB and EV:
-WB send the last Cable Check Response (or PreCharge Response) with ResponseCode=OK to the EV, including EVSEStatusCode=EVSE_ShutDown and EVSENotification=StopCharging
-EV send Session Stop Request
-WB answers with Session Stop Response with ResponseCode=FAILED_SequenceError
As a result, the EV does not release the connector immediately, but within about 15 seconds.

Here is the message sequence between WB and the host Controller (for user stop during Cable Check, for PreCharge the sequence is similar):
ERx: 123456789ABC C49300215136 6003 0004000E C02787FF0006 00000609004C6DC1 -Request Cable Check Parameters
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000B C0276D170003 000000CEC1 -Set Cable Check Parameters (Resultcode=EV parameters accepted)
ERx: 123456789ABC C49300215136 6003 00040009 C0276D170001 00D0C1
Here is the stop request from user                                                                             
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000A C02944180002 03E80AC1 -Set PWM=100%
ERx: 123456789ABC C49300215136 6003 00040009 C02944180001 00F6C1
ERx: 123456789ABC C49300215136 6003 0004000E C02787FF0006 00000608004C6EC1 -Request Cable Check Parameters
ETx: FFFFFFFFFFFF 123456789ABC 6003 0004000D C0276D190005 0300000000C7C1 -Set Cable Check Parameters (Resultcode=Stop Charging)
ERx: 123456789ABC C49300215136 6003 00040009 C0276D190001 00CEC1
ERx: 123456789ABC C49300215136 6003 00040009 C02981FF0001 01D0C1 -CPState=B
ERx: 333300000002 007DFA03950A 86DD
ERx: 123456789ABC C49300215136 6003 00040008 C02781FF0000 D4C1  -Session Stopped

Can you please check why the ResponseCode=FAILED_SequenceError is sent to EV in the Session Stop Response message?

Thank You,
Ionel Fratila

**Comment by IonelFratila** [comment]
Hi @jpo-stx ,
The test was using ISO15118 defined by Set supported protocols command (SubID=0x60), but the requirements are very similar for both DIN SPEC 70121 and ISO15118 regarding the sequence of the messages between EV and EVSE. The normal sequence is not described according to these standards exactly as you said (please review DIN SPEC 70121 Chapter 9.7.4. Request-response message sequence requirements). Here are some of these requirements ([V2G-DC-648] and [V2G-DC 650] refer exactly to our situation):

**[V2G-DC 390]** The response message shall contain the ResponseCode "FAILED_SequenceError" if the SECC has received an unexpected request message.
**9.7.4.1.2.EVCC requirements for premature intentional termination**
**[V2G-DC-648]** If the EVCC wishes to stop the Communication Session for a non critical reason (e.g.user interaction) after having sent a SessionSetupReq message and before having sent a PowerDeliveryReq message with ReadyToCharge state equal to "TRUE", the EVCC should not imediatelly change the CP State, unless this is required by another requirement in this document, but EVCC shall send a SessionStopReq message as the next request message and then wait for a SessionStopRes message. It shall then proceed  according to [V2G-DC413], [V2G-DC 750], [V2G-DC 657] and [V2G-DC 658].
**[V2G-DC 650]** If the EVCC receives a response message that it expects in response to the latest request message sent, and this response message contains EVSEStatusCode equal to "EVSE_Shutdown", the EVCC shall proceed as if the EVCC itself wished to stop the Communication Session for a non critical reason according to [V2G-DC 648] and [V2G-DC 673].

In conclusion, it is perfectly normal according to DIN SPEC 70121 (similar requirements in ISO15118) for EV to send a SessionStopReq message right after a CableCheckReq (or PreChargeReq) message. To this message EVSE must answer with a SessionStopRes message with ResponseCode=OK. The SessionStopReq message is not a SequenceError in this case. The message sequence Power delivery stop -> Welding detection -> Session stop that you described is sent by EV only after a PowerDeliveryReq confirmed message (meaning after PreCharge), according to [V2G-DC 673].

Please check and correct the WB SW according to the requirements of the standards.
Thank You,
Ionel Fratila

**Comment by IonelFratila** [comment]
Hi @jpo-stx ,
Please let me know if you had time to analyze and if you have an answer to my previous questions:

1.Accept the SessionStopReq message sent by EV (and answer with ResponseCode=OK), even the request is sent before PowerDeliveryReq (ReadyToCharge=True), meaning during Cable Check or Precharge stage (DIN 701201 and ISO15118 requirements)
2.Why WB is not using the same values ​​for NID and NMK parameters as at the beginning of the charging session (first SLAC Matching process after EV plug-in) for subsequent SLAC matching runs, until EV is plug-out (mandatory requirement of the standards)?
3.How and when the command Join HPGP network (ModID=0x2A, SubID=0x40) can be used to send particular values for NID and NMK to WB (every time I get the result code=0x01 (Module Busy?!) ?.

Thank You for your time,
Ionel Fratila

**Comment by jpo-stx** [comment]
Hi @IonelFratila ,
sorry for taking so long.
1. You are correct. We need to change the handling of the SessionStop for the DIN protocol! We are planning this for the next firmware release.

@bbr-stx can you please comment the points 2 and 3

**Comment by bbr-stx** [comment]
Hi @IonelFratila,

   2. Currently the sleep mode is not supported. Unfortunately, I can't say yet when we will deliver this feature.

   3. I have added this issue to our ticket system. As soon as we have new information about this, I will announce it here.

**Comment by IonelFratila** [comment]
Hi @jpo-stx and @bbr-stx ,
Thank you for the answers.
1. Next SW release will be V.01_01_07 or the next one?
2.I don't see the connection between keeping the NID and NMK unmodified from EV plug-in to EV plug-out (there are precise  requirements for this, at least in DIN 70121, but I think I have read the same also in ISO15118, I will check again) and the sleep mode (the requirements for NID and NMK are not referring to this sleep mode in any way).
Meanwhile I will wait for the next SW release.
Thank you,
BR
Ionel Fratila

**Comment by bbr-stx** [comment]
Hi @IonelFratila ,

point 2. will be included in the next release (PnC).

On the charging station side, the 'Start SLAC Matching' command can then be called again and the old NID and NMK are then used for the SLAC Mathing process. The host can recognize by the BCB toggle on CP that the SLAC matching wants to be executed again by the vehicle.

**Comment by IonelFratila** [comment]
Hi @jpo-stx ,
Because I'm testing now this issue on the new version WB V.02_00_00, I will close this and I will open another issue depending on the results.
Thank You,
Ionel Fratila
